augur/merge.py

```diff
@@ -503,3 +579,85 @@ def quote(s):
     # …and instead quote a final empty string down here if we're still empty
     # after joining all our parts together.
     return quoted if quoted else shquote('')
+
+
+def run_command(command: str, print_stdout: bool):
+    """Run a command, clearly marking any messages that resemble SeqKit output."""
+    process = subprocess.Popen(
+        ['bash', '-euo', 'pipefail', '-c', command],
+        stdout=subprocess.PIPE if print_stdout else subprocess.DEVNULL,
+        stderr=subprocess.PIPE,
+        text=True,
+    )
+
+    if print_stdout:
+        for line in iter(process.stdout.readline, ''):
```

Comment:
Why not

```suggestion
        for line in process.stdout:
```

Is there something about the `iter()` approach above that I'm missing? 

(and ditto for stderr below)

Reply:
I think I ended up with the `iter()` approach in an attempt to avoid deadlock issues, but not sure if it actually helped.

This is no longer applicable with the switch to `run_shell_command` in https://github.com/nextstrain/augur/pull/1631#discussion_r2008282952